### PR TITLE
add deploy-related files

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

### CODE REVIEW

File: deploy/config/anp/anp_deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          ports:
+            - name: userport
+              containerPort: 9092
+      volumes:
+        - name: tls-vol
+          secret:
+            secretName: cluster-proxy-addon-serving-cert
+        - name: ca-vol
+          configMap:
+            name: cluster-proxy-ca-bundle
```
Comment (skeeey) [inline deploy/config/anp/anp_deployment.yaml:63]: need a new line

File: deploy/config/anp/service.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: cluster-proxy-service
+spec:
+  ports:
+    - name: agentport
+      port: 9091
+    - name: userport
+      port: 9092
+  selector:
+    app: anp-server
```
Comment (skeeey) [inline deploy/config/anp/service.yaml:12]: need a new line

File: deploy/config/operator/operator.yaml [hunk 3]
```diff
@@ -38,9 +38,4 @@ spec:
         resources:
           requests:
             cpu: 100m
-            memory: 128Mi
-        env:
```
Comment (skeeey) [inline deploy/config/operator/operator.yaml:42]: why delete this? I think we should keep this

File: deploy/config/operator/operator.yaml [hunk 4]
```diff
@@ -38,9 +38,4 @@ spec:
...
-          - name: POD_NAME
-            valueFrom:
-              fieldRef:
-                fieldPath: metadata.name
+            memory: 128Mi
```
Comment (skeeey) [inline deploy/config/operator/operator.yaml:41]: need a new line

File: pkg/hub/addon/manifests/anp_deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            - name: ca-vol
+              mountPath: /ca
+              readOnly: true
+      volumes:
+        - name: tls-vol
+          secret:
+            secretName: cluster-proxy-open-cluster-management.io-cluster-proxy-addon-client-cert
+        - name: ca-vol
+          configMap:
+            name: cluster-proxy-ca-bundle
```
Comment (skeeey) [inline pkg/hub/addon/manifests/anp_deployment.yaml:60]: need a new line

File: pkg/hub/addon/manifests/clusterrolebinding.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cluster-proxy-agent
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
```
Comment (skeeey) [inline pkg/hub/addon/manifests/clusterrolebinding.yaml:8]: we need a fine-grained clusterrole

File: pkg/hub/addon/manifests/anp_deployment.yaml (hunk 5 above)
Comment (skeeey) [inline pkg/hub/addon/manifests/anp_deployment.yaml:60]: do we need a template form cm？

File: pkg/hub/addon/manifests/addon_deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    spec:
+      serviceAccountName: cluster-proxy-agent-sa
+      volumes:
+        - name: hub-config
+          secret:
+            secretName: {{ .KubeConfigSecret }}
+      containers:
+        - name: cluster-proxy-agent
+          image: {{ .Image }}
+          imagePullPolicy: IfNotPresent
```
Comment (skeeey) [inline pkg/hub/addon/manifests/addon_deployment.yaml:26]: remove this, or set this to always

File: pkg/hub/addon/manifests/anp_deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        app: anp-agent
+    spec:
+      hostAliases:
+        - ip: "127.0.0.1"
+          hostnames:
+            - "{{ .ClusterName }}"
+      containers:
+        - name: apiserver-proxy
+          image: {{ .Image }}
+          imagePullPolicy: IfNotPresent
```
Comment (skeeey) [inline pkg/hub/addon/manifests/anp_deployment.yaml:24]: remove this or set this to always

File: pkg/hub/addon/manifests/anp_deployment.yaml (hunk 5 above)
Comment (xuezhaojun) [reply pkg/hub/addon/manifests/anp_deployment.yaml:60]: This configmap is create by controller

File: pkg/hub/addon/manifests/clusterrole.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: open-cluster-management:cluster-proxy-addon:agent
+rules:
+# Allow submariner-addon agent to run with openshift library-go
```
Comment (skeeey) [inline pkg/hub/addon/manifests/clusterrole.yaml:6]: rename the comments

File: pkg/hub/addon/manifests/clusterrole.yaml [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs: ["get", "list", "watch"]
+# Allow submariner-addon agent to get cloud credations
+- apiGroups: [""]
```
Comment (skeeey) [inline pkg/hub/addon/manifests/clusterrole.yaml:11]: we don't need this

File: pkg/hub/addon/manifests/clusterrole.yaml [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+  resources: ["secrets"]
+  verbs: ["get"]
+# Allow submariner-addon agent to get/update nodes
+- apiGroups: [""]
```
Comment (skeeey) [inline pkg/hub/addon/manifests/clusterrole.yaml:15]: we don't need this

File: pkg/hub/addon/manifests/role.yaml [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+- apiGroups: ["", "events.k8s.io"]
+  resources: ["events"]
+  verbs: ["create", "patch", "update"]
+# Allow cluster-proxy-addon agent to run with addon-framwork
+- apiGroups: ["coordination.k8s.io"]
+  resources: ["leases"]
+  verbs: ["get", "list", "watch", "create", "update", "delete"]
+# Allow cluster-proxy-addon agent to monitor cluster-proxy deployment status
+- apiGroups: ["operators.coreos.com"]
+  resources: ["subscriptions"]
```
Comment (skeeey) [inline pkg/hub/addon/manifests/role.yaml:26]: we don't need this

File: pkg/hub/addon/manifests/role.yaml [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["apps"]
+  resources: ["daemonsets", "deployments"]
```
Comment (skeeey) [inline pkg/hub/addon/manifests/role.yaml:29]: we don't need this now

**Pull request merged**